### ISSUE TITLE
Docker getting-started guide is broken

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#10962]
How to reproduce:

```
cd examples/getting-started
vagrant up
vagrant ssh
docker network create --driver cilium --ipam-driver cilium cilium-net
docker run -d --name app1 --net cilium-net -l "id=app1" cilium/demo-httpd
docker run --rm -ti --net cilium-net -l "id=app2" cilium/demo-client curl -m 20 http://app1
Unable to find image 'cilium/demo-client:latest' locally
latest: Pulling from cilium/demo-client
605ce1bd3f31: Pull complete
3ec819207123: Pull complete
Digest: sha256:311992b23fe8977472bd73ad9a45f05c923fbf90b38e6c512a7537c2ffda75ca
Status: Downloaded newer image for cilium/demo-client:latest
curl: (28) Connection timed out after 20001 milliseconds
```

Diagnosis:

Container is running:
```
docker ps -a
CONTAINER ID        IMAGE                         COMMAND                  CREATED             STATUS              PORTS                                                                                NAMES
8ea28f36a3de        cilium/demo-httpd             "httpd-foreground"       44 seconds ago      Up 43 seconds                                                                                            app1
dd406d9870e9        cilium/docker-plugin:v1.7.0   "cilium-docker"          2 minutes ago       Up 2 minutes                                                                                             cilium-docker-plugin
7afa028ec57c        cilium/cilium:v1.7.0          "cilium-agent --enab…"   2 minutes ago       Up 2 minutes                                                                                             cilium
b7197a9a9aab        consul:0.8.3                  "docker-entrypoint.s…"   2 minutes ago       Up 2 minutes        8300-8302/tcp, 8400/tcp, 8301-8302/udp, 8600/tcp, 8600/udp, 0.0.0.0:8500->8500/tcp   cilium-kvstore
```

Container is not part of `cilium endpoint list`
```
cilium endpoint list
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])   IPv6   IPv4           STATUS
           ENFORCEMENT        ENFORCEMENT
2090       Enabled            Enabled           5          reserved:init                        10.15.239.91   ready
3328       Disabled           Disabled          4          reserved:health                      10.15.150.21   ready
```

`docker inspect ...`
```
[...]
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "9d7d83da1c0d795fb5dc4bace50b1434938194a922cb67afd42a61d3efd5a015",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/9d7d83da1c0d",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "cilium-net": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "8ea28f36a3de"
                    ],
                    "NetworkID": "797a8f75ca82af8442bb336019052430bb4afd48c4da68322486ca0543a74e97",
                    "EndpointID": "677094a0c2ac0389fae7a04c91587caee989ec34a609a840156d05b56bbaa631",
                    "Gateway": "",
                    "IPAddress": "10.15.239.91",
                    "IPPrefixLen": 32,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

**Comment by tklauser** [comment]
`docker logs` of the `cilium/docker-plugin` shows:

```
level=info msg="Waiting for cilium daemon to start up..." ciliumSockPath="unix:///var/run/cilium/cilium.sock" dockerHostPath="unix:///var/run/docker.sock" subsys=cilium-docker-driver
level=info msg="Waiting for cilium daemon to start up..." ciliumSockPath="unix:///var/run/cilium/cilium.sock" dockerHostPath="unix:///var/run/docker.sock" subsys=cilium-docker-driver
level=info msg="Waiting for cilium daemon to start up..." ciliumSockPath="unix:///var/run/cilium/cilium.sock" dockerHostPath="unix:///var/run/docker.sock" subsys=cilium-docker-driver
level=info msg="Waiting for cilium daemon to start up..." ciliumSockPath="unix:///var/run/cilium/cilium.sock" dockerHostPath="unix:///var/run/docker.sock" subsys=cilium-docker-driver
level=info msg="Waiting for cilium daemon to start up..." ciliumSockPath="unix:///var/run/cilium/cilium.sock" dockerHostPath="unix:///var/run/docker.sock" subsys=cilium-docker-driver
level=info msg="Starting docker events watcher" subsys=cilium-docker-driver
level=info msg="Cilium Docker plugin ready" subsys=cilium-docker-driver
level=info msg="Listening for events from Docker" file-path=/run/docker/plugins/cilium.sock subsys=cilium-docker
level=error msg="Unable to connect to docker events channel, reconnecting..." error="Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?" subsys=cilium-docker-driver
level=error msg="Unable to connect to docker events channel, reconnecting..." error="Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?" subsys=cilium-docker-driver
level=error msg="Unable to connect to docker events channel, reconnecting..." error="Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?" subsys=cilium-docker-driver
[...]
```

**Comment by tklauser** [comment]
Fix in #11023, bind mount for `/var/run/docker.sock` was missing for the `cilium-docker` container.
